enhancements/kube-controller-manager/max-namespace-size.md

```diff
@@ -0,0 +1,112 @@
+---
+title: limit-resources-per-namespace
+authors:
+  - "@deads2k"
+reviewers:
+  - "@soltysh"
+approvers:
+  - "@mfojtik"
+creation-date: 2020-01-31
+last-updated: 2020-01-31
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Limit Resources Per Namespace
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a new `namespace-size-policy-controller` in the existing [cluster-policy-controller](https://github.com/openshift/cluster-policy-controller/).
+This controller will create quota objects in every namespace that limit the size of certain error-prone resources.
+We will start with secrets and configmaps.
+It will be created with a goal of protecting the cluster from accidental runaway resource creation.
```

Comment:
If the goal is to protect etcd and the cluster shouldn't we apply the generic `count/*` quota? Then possibly limit the subranges for particular resources?